### PR TITLE
D2IQ-69562: empty state guidelines

### PR DESCRIPTION
**Pull request by mperrotti**
Closes D2IQ-69562

## Testing

Read the guidelines about how we handle cases where content cannot be shown.

Please comment if:

- anything in the guidelines is unclear
- you can think of scenarios that are not covered by these guidelines
- you disagree with anything from these guidelines

## Checklist for PR creator

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [x] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [x] Info for applicable sections above is provided

### CODE REVIEW

File: design-guidelines/empty-state-patterns.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+For example: if a page has loaded everything except for the data required to render a table, we should not replace the entire screen with a loading indicator. We can at least show the page header and the app's chrome.
+
+#### Loading inline content
+
+When waiting for data to show a small piece of data, show all of the loaded content and show a loading indicator in place of the content that is still being loaded.
+
+For example: showing a loading indicator in place of a package version number until the API call to get the package data.
+
+#### Loading data needed to take an action
```
Comment (danielone) [inline design-guidelines/empty-state-patterns.md:69]: Header feels a bit unclear. 
comes off as data needs to load in order to take action vs. action on interactive component has triggered action.

File: design-guidelines/empty-state-patterns.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+What to do
+
+- Do not block unrelated content from being shown. Show as much content as possible while waiting for the full response
+- If the delay is especially long and there is an estimate for how long it will take, share that time estimate with the user
+- If a wait time is especially long because there are multiple steps that need to be taken, let users know what step they're on. If possible, inform the user how many more steps there are
+- Show a loading indicator in place of the content being loaded
+
+### When to show a loading indicator
+
+To helped improve the perceived performance of our apps, don't let loading indicators hide available content. Loading indicators are only displayed in place of content that cannot possibly be rendered until some action or request has been processed.
```
Comment (pierrebeitz) [inline design-guidelines/empty-state-patterns.md:53]:
```suggestion
To improve the perceived performance of our apps, don't let loading indicators hide available content. Loading indicators are only displayed in place of content that cannot possibly be rendered until some action or request has been processed.
```

File: design-guidelines/empty-state-patterns.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+#### Loading an entire screen or large sections of a screen
+
+When an entire section of a page can't be displayed, show a loading indicator in place of the loading content.
+
+Avoid showing more than one section-level loading indicator. Multiple sections in a "loading" state could hurt perceived performance by making the user feel like there are a lot of large chunks of data to wait for.
+
+For example: if a page has loaded everything except for the data required to render a table, we should not replace the entire screen with a loading indicator. We can at least show the page header and the app's chrome.
+
+#### Loading inline content
```
[the remark this answers is not in the document]
Comment (pierrebeitz) [reply design-guidelines/empty-state-patterns.md:63]: > no camel case

?

> just inline

?

> Loading inlilne or section level content

are you saying that we should make a difference between a "large" section and a regular section? i feel like the section level is handled in the paragraph above.

> inlline/field or cell level good to callout at section level - under a specific header - is important to call out.

are you suggesting to rearrange this document? in what way exactly?

**Review by pierrebeitz — DISMISSED** [review]

File: design-guidelines/empty-state-patterns.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mperrotti) [reply design-guidelines/empty-state-patterns.md:63]: @danielone - I have all the same questions as Pierre 😅

File: design-guidelines/empty-state-patterns.md (hunk 1 above)
Comment (mperrotti) [reply design-guidelines/empty-state-patterns.md:69]: > comes off as data needs to load in order to take action

Good, that's exactly what I meant :)

**Pull request merged**